fdbclient/S3Client.actor.cpp

```diff
@@ -123,86 +138,218 @@ Reference<S3BlobStoreEndpoint> getEndpoint(const std::string& s3url,
 		    S3BlobStoreEndpoint::fromString(s3url, proxy, &resource, &error, &parameters);
 
 		if (!endpoint) {
-			TraceEvent(SevError, "S3ClientGetEndpointNullEndpoint").detail("URL", s3url).detail("Error", error);
+			TraceEvent(SevError, "S3ClientGetEndpointNullEndpoint")
+			    .detail("URL", StringRef(s3url))
```

Comment:
Why do we need `StringRef(s3url)` to print s3url? Thanks

Reply:
Save a string copy?

Reply:
Seems that StringRef just hold a pointer to s3url? Please correct me if I am wrong.